File: android/vbt-app/app/src/main/java/com/vbt/app/domain/usecase/GetVelocityZoneUseCase.kt
package com.vbt.app.domain.usecase

import com.vbt.app.domain.model.VelocityZone
import javax.inject.Inject

class GetVelocityZoneUseCase @Inject constructor() {
    operator fun invoke(velocity: Float): VelocityZone {
        return VelocityZone.fromVelocity(velocity)
    }
}

File: ios/VBT/VBT/Domain/UseCases/GetVelocityZoneUseCase.swift
import Foundation

/// Port 1:1 z Android `domain/usecase/GetVelocityZoneUseCase.kt`.
struct GetVelocityZoneUseCase {
    func callAsFunction(_ velocity: Float) -> VelocityZone {
        VelocityZone.from(velocity: velocity)
    }
}
